pyart/io/nexrad_level2.py

```diff
@@ -621,11 +622,11 @@ def _get_msg31_data_block(buf, ptr):
         dic = _unpack_from_buf(buf, ptr, ELEVATION_DATA_BLOCK)
     elif block_name == 'RAD':
         dic = _unpack_from_buf(buf, ptr, RADIAL_DATA_BLOCK)
-    elif block_name in ['REF', 'VEL', 'SW', 'ZDR', 'PHI', 'RHO']:
+    elif block_name in ['REF', 'VEL', 'SW', 'ZDR', 'PHI', 'RHO', 'CFP']:
         dic = _unpack_from_buf(buf, ptr, GENERIC_DATA_BLOCK)
         ngates = dic['ngates']
         ptr2 = ptr + _structure_size(GENERIC_DATA_BLOCK)
-        if block_name == 'PHI':
+        if block_name == 'PHI' or block_name == 'ZDR':
```

Comment:
There's a field in the message that explicitly states the size of each bin in bits. Could you not just use that rather than hard-coding it here, which just puts you back in the same position if they update again?

Reply:
That makes sense, and sounds good to me. I will do some changes. Thanks for the feedback!

Reply:
@dopplershift Just so I got this right as I'm starting to confuse myself lol, still set dtype for the moments, but use something like int16 and not hard coding to set a specific moment?

Reply:
Not quite. Here's a section of the specification on the Data Block:

<img width="691" alt="image" src="https://user-images.githubusercontent.com/221526/74390244-e8e99000-4dbd-11ea-96df-87517df34211.png">

Note the field "Data Word Size", it can (currently at least) have a value of 8 or 16. I'm proposing using that value to decide which dtype code is appropriate. [This](https://github.com/Unidata/MetPy/blob/40d5c12ab341a449c9398508bd41d010165f9eeb/src/metpy/io/_tools.py#L313-L321) is the code we use to do that in MetPy.

Reply:
Ahhh gotcha, that makes sense. Yeah the bit of code in that link helps clear things up. Thanks for helping clarify that!

Reply:
No problem!